### PR TITLE
[21561] Bump version to 3.0.1

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR bumps `ShapesDemo` to `v3.0.1`

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- **N/A** Changes do not break current interoperability.
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- **N/A** CI passes without warnings or errors.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,7 +125,7 @@ option(EPROSIMA_BUILD "Activate internal building" OFF)
 ###############################################################################
 # Load external eProsima projects.
 ###############################################################################
-set(FASTDDS_MIN_VERSION "3.0.0")
+set(FASTDDS_MIN_VERSION "3.0.1")
```
Comment (Mario-DL) [inline CMakeLists.txt:128]: I doubt here whether we should keep `3.0.0`

Comment (JesusPoderoso) [reply CMakeLists.txt:128]: It is not part of the bumping process.
I guess we updated it due to the major version, but I would keep it as it was (`3.0.0`).

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Comment by MiguelCompany** [comment]
We usually update the .repos files when we do the bump

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**